src/storage/mapping/ExtensionBasedMapper.ts

```diff
@@ -1,18 +1,27 @@
 import { promises as fsPromises } from 'fs';
 import * as mime from 'mime-types';
 import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
-import { TEXT_TURTLE } from '../../util/ContentTypes';
+import { APPLICATION_TRIG, TEXT_TURTLE } from '../../util/ContentTypes';
 import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
 import { joinFilePath, getExtension } from '../../util/PathUtil';
 import { BaseFileIdentifierMapper } from './BaseFileIdentifierMapper';
 import type { FileIdentifierMapperFactory, ResourceLink } from './FileIdentifierMapper';
 
 export class ExtensionBasedMapper extends BaseFileIdentifierMapper {
   private readonly types: Record<string, any>;
+  private readonly mimeFallback: Record<string, string>;
 
-  public constructor(base: string, rootFilepath: string, overrideTypes = { acl: TEXT_TURTLE, meta: TEXT_TURTLE }) {
+  public constructor(
+    base: string,
+    rootFilepath: string,
+    overrideTypes: Record<string, string> = { acl: TEXT_TURTLE, meta: TEXT_TURTLE, trig: APPLICATION_TRIG },
+  ) {
     super(base, rootFilepath);
     this.types = { ...mime.types, ...overrideTypes };
+    this.mimeFallback = {};
```

Comment:
```suggestion
    this.extensions = { ...mime.types, ...overrideTypes };
    this.contentTypes = {};
```

How come we use `mime.types` above but `mime.extension(…)` below? Can't we use the same mechanism?

Reply:
Probably because when this was originally written we only were interested in changing 1 direction (extension -> type). The reverse wasn't relevant for acl/meta.

I had a look at the typings and they also expose a `mime.extensions` object so we could do the same there. Or we can change types object to be similar to how the extensions were solved here to make sure we don't have to duplicate these large objects in memory.